lib/puppet/provider/cisco_itd_service/cisco.rb

```diff
@@ -0,0 +1,256 @@
+#
+# The Cisco provider for cisco_itd_device_group.
+#
+# March 2016
+#
+# Copyright (c) 2016 Cisco and/or its affiliates.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require 'cisco_node_utils' if Puppet.features.cisco_node_utils?
+begin
+  require 'puppet_x/cisco/autogen'
+rescue LoadError # seen on master, not on agent
+  # See longstanding Puppet issues #4248, #7316, #14073, #14149, etc. Ugh.
+  require File.expand_path(File.join(File.dirname(__FILE__), '..', '..', '..',
+                                     'puppet_x', 'cisco', 'autogen.rb'))
+end
+
+begin
+  require 'puppet_x/cisco/cmnutils'
+rescue LoadError # seen on master, not on agent
+  # See longstanding Puppet issues #4248, #7316, #14073, #14149, etc. Ugh.
+  require File.expand_path(File.join(File.dirname(__FILE__), '..', '..', '..',
+                                     'puppet_x', 'cisco', 'cmnutils.rb'))
+end
+Puppet::Type.type(:cisco_itd_service).provide(:cisco) do
+  desc 'The Cisco provider for cisco_itd_service.'
+
+  confine feature: :cisco_node_utils
+  defaultfor operatingsystem: :nexus
+
+  mk_resource_methods
+
+  ITDSERVICE_NON_BOOL_PROPS = [
+    :access_list,
+    :device_group,
+    :exclude_access_list,
+    :load_bal_buckets,
+    :load_bal_mask_pos,
+    :load_bal_method_bundle_hash,
+    :load_bal_method_bundle_select,
+    :load_bal_method_end_port,
+    :load_bal_method_proto,
+    :load_bal_method_start_port,
+    :peer_local,
+  ]
+  ITDSERVICE_BOOL_PROPS = [
+    :fail_action,
+    :load_bal_enable,
+    :nat_destination,
+  ]
+  # shutdown property is treated separately due to the ordering.
+  # When shutdown goes from true to false, it needs to be set
+  # after all the other properties are set. For all other cases,
+  # shutdown needs to be set first before any other
+  # properties are set. Basically, no properties cannot be
+  # changed while the service is active.
+  ITDSERVICE_SHUT_PROP = [
```

Comment:
Instead of creating a separate special array for this, just do the following:
- Add it to `ITDSERVICE_BOOL_PROPS` array so that getters and setters get autogenerated for all bool props.
- In the `initialize` method, remove `:shutdown` from the `ITDSERVICE_BOOL_PROPS` array.